### ISSUE TITLE
Reinstall of Fedora CoreOS after adding disk bombs

### ISSUE THREAD

**Issue by shivarammysore** [coreos/fedora-coreos-tracker#282]
I used `mmcblk0` as the initial boot disk and was able to get Fedora CoreOS installed and fully operational.  I then added a disk `sda`.  I reinstalled (via PXE - fully wipe) coreos on `mmcblk0`

Generating "/run/initramfs/rdsosreport.txt"

```
Entering emergency mode. Exit the shell to continue.
Type "journalctl" to view system logs.
You might want to save "/run/initramfs/rdsosreport.txt" to a USB stick or /boot
after mounting them and attach it to a bug report.
```

possible culprit segment: (attached full report)

```
0m] Started dracut initqueue hook.
[   19.087502] audit: type=1130 audit(1569089886.588:10): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=dracut-initqueue comm="systemd'
[   19.087975] systemd[1]: Reached target Remote File Systems (Pre).
[  OK  ] Reached target Remote File Systems (Pre).
[  OK     19.122542] systemd[1]: Reached target Remote File Systems.
0m] Reached target Remote File Systems.
[   19.139331] systemd[1]: Starting dracut pre-mount hook...
         Starting dracut pre-mount hook...
[   19.187689] systemd[1]: Started dracut pre-mount hook.
[  OK  ] Started dracut pre-mount hook.
[   19.201538] audit: type=1130 audit(1569089886.702:11): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=dracut-pre-mount comm="systemd'
[   19.204533] systemd[1]: Starting File System Check on /dev/disk/by-label/root...
         Starting File System Check on /dev/disk/by-label/root...
[   19.260857] systemd-fsck[863]: /usr/sbin/fsck.xfs: XFS file system.
[  OK     19.270410] systemd[1]: Started File System Check on /dev/disk/by-label/root.
0m] Started File System Check on /dev/disk/by-label/root.
[   19.291614] systemd[1]: Mounting /sysroot...
         Mountin[   19.296078] audit: type=1130 audit(1569089886.789:12): pid=1 uid=0 auid=4294967295 ses=4294967295 subj=kernel msg='unit=systemd-fsck-r'
g /sysroot...
[   19.771792] SGI XFS with ACLs, security attributes, scrub, no debug enabled
[   19.788579] XFS (sda4): Mounting V5 Filesystem
[   19.813813] XFS (sda4): Ending clean mount
[  OK     20.125540] systemd[1]: Mounted /sysroot.
0m] Mounted /sysroot.
[   20.137017] systemd[1]: Condition check resulted in Remount /sysroot read-write for Ignition being skipped.
         Startin[   20.147072] systemd[1]: Starting OSTree Prepare OS/...
g OSTree Prepare OS/   20.155340] ostree-prepare-root[885]: ostree-prepare-root: Couldn't find specified OSTree root '/sysroot//ostree/boot.1/fedora-corey
m...
[   19.202077] ostree-prepare-root[885]: ostree-prepare-root: Couldn't find specified OSTree root '/sysroot//ostree/boot.1/fedora-coreos/b2601a3ea2062ef1y
[FAILED[   20.201741] systemd[1]: ostree-prepare-root.service: Main process exited, code=exited, status=1/FAILURE
] Failed to [   20.212566] systemd[1]: ostree-prepare-root.service: Failed with result 'exit-code'.
start O[   20.221776] systemd[1]: Failed to start OSTree Prepare OS/.
STree Prepare OS/
```

Let me know if you need any other info

[rdsosreport.txt](https://github.com/coreos/fedora-coreos-tracker/files/3640293/rdsosreport.txt)

**Comment by dustymabe** [comment]
You are running a fresh install via PXE? Can you share more information? 

What was on `sda`?

**Comment by jlebon** [comment]
> [   19.202077] ostree-prepare-root[885]: ostree-prepare-root: Couldn't find specified OSTree root '/sysroot//ostree/boot.1/fedora-coreos/b2601a3ea2062ef1y

At first I thought this was likely because there were two filesystems with the `root` label. Though looking at the sosreport:

```
+ blkid
/dev/sda1: LABEL="boot" UUID="c8d4731a-9c1f-4f5b-a4ec-1f03526bf3aa" TYPE="ext4" PARTLABEL="boot" PARTUUID="1a00f529-2d86-4284-99ac-a97c993eeaaf"
/dev/sda2: SEC_TYPE="msdos" LABEL_FATBOOT="EFI-SYSTEM" LABEL="EFI-SYSTEM" UUID="1891-EF1E" TYPE="vfat" PARTLABEL="EFI-SYSTEM" PARTUUID="9b8f4db6-f6c4-437"
/dev/sda3: PARTLABEL="BIOS-BOOT" PARTUUID="0f04ae23-72ad-4627-8b75-b9aba1f57675"
/dev/sda4: LABEL="root" UUID="3dc48ff8-d734-432e-95b6-0a47308fddf3" TYPE="xfs" PARTLABEL="root" PARTUUID="77b866be-9d57-450e-a8f7-ce3214ffb555"
/dev/mmcblk0: PTUUID="837f99a0-8c90-435e-af99-35f53eabc1eb" PTTYPE="gpt"
/dev/mmcblk0p1: LABEL="boot" UUID="c8d4731a-9c1f-4f5b-a4ec-1f03526bf3aa" TYPE="ext4" PARTUUID="065d8184-7075-49b1-8b41-45037f4b4eae"
```

`/dev/sda4` is root. But you were trying to target `/dev/mmcblk0` during the install?

Hmm, kernel cmdline shows `root=/dev/disk0`, which seems... odd. That should normally be`root=/dev/disk/by-label/root`. Did you modify that kernel arg?

**Comment by shivarammysore** [comment]
It was a fresh install via PXE.  @jlebon observation is correct.  My PXE `default` file has: 

```
KERNEL fedora-coreos-30.20190801.0-installer-kernel
  APPEND ip=dhcp rd.neednet=1 initrd=fedora-coreos-30.20190801.0-installer-initramfs.img console=tty0 console=ttyS0,115200n8 coreos.inst=yes coreos.inst.text=yes coreos.inst.install_dev=mmcblk0 coreos.autologin=tty1 coreos.autologin=ttyS0 coreos.inst.image_url=http://10.20.30.2:8000/fedora-coreos-30.20190801.0-metal.raw.xz coreos.inst.ignition_url=http://10.20.30.2:8000/bare_metal_ignition.json
  IPAPPEND 2
```

What I observed is that `mmcblk0` does not get wiped clean.  I had to disk nuke it separately to get installed again.  It was lots of frustration.  When you have more than one disk, it automatically moves the boot partition to the other one too (more testing is needed - observed one).

I just want `boot`, `root` and `swap` partitions on a single disk.  No extra kernel args were added - only the PXE options as noted above.

**Comment by dustymabe** [comment]
@jlebon, you labeled this as a bug on the installer.. do we know what needs to be done in the installer to resolve this?

**Comment by jlebon** [comment]
@dustymabe I'm not sure offhand, we'd have to reproduce it.

@shivarammysore Is the error in https://github.com/coreos/fedora-coreos-tracker/issues/282#issue-496849741 from the first boot after the re-install?

>  When you have more than one disk, it automatically moves the boot partition to the other one too (more testing is needed - observed one).

I'm not sure how the installer would do this -- it literally just `dd`s the full image to the target block device. Can you sanity check the output of `fdisk -l dev/mmcblk0` after the install?

**Comment by shivarammysore** [comment]
@jlebon - yes the error is from first boot after re-install.  

From the error messages during install I see that it was erasing about first 6[8] bytes (a single digit number - I forget the exact one) on the disk.  There were no messages that it was wiping down the device.

My machine now is just using `mmcblk0` as I had to get work done and could not keep it in that state - I basically decided to just use 1 disk!

May be a question/suggestion - how can we set a device as the main device on which `/boot`, `/var` and all partitions must be present?  Can we also set which other devices must be wiped down - all of these as PXE options.  Sending it via ignition will be too late as the boot loader will already be installed.

This is a problem and I lost lot of time trying to get my system recovered from this.
